src/Events/Integrations/Modules/Tickets_Emails/Email/RSVP.php

```diff
@@ -0,0 +1,179 @@
+			'label'           => esc_html__( 'Include "Add to calendar" links', 'the-events-calendar' ),
+			'tooltip'         => esc_html__( "Include links to add the event to the user's calendar.", 'the-events-calendar' ),
+			'default'         => true,
+			'validation_type' => 'boolean',
+		];
+
+		$settings[ static::$option_add_event_ics ] = [
+			'type'            => 'toggle',
+			'label'           => esc_html__( 'Attach calendar invites', 'the-events-calendar' ),
+			'tooltip'         => esc_html__( 'Attach calendar invites (.ics) to the RSVP email.', 'the-events-calendar' ),
+			'default'         => true,
+			'validation_type' => 'boolean',
+		];
+
+		return $settings;
+	}
+
+	/**
+	 * Filters the attachments for the RSVP Emails and maybe add the calendar ics file.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,string> $attachments  The attachments for the Tickets Emails.
+	 * @param string               $email_id     The email ID.
+	 * @param Email_Abstract       $email_class  The email class.
+	 *
+	 * @return array<string,string> The filtered attachments for the RSVP Emails.
+	 */
+	public function include_attachments( $attachments, $email_id, $email_class ) {
+		if ( ! $email_class->is_enabled() ) {
+			return $attachments;
+		}
+
+		$use_ticket_email = tribe_get_option( $email_class->get_option_key( 'use-ticket-email' ), false );
+
+		if ( ! empty( $use_ticket_email ) ) {
+			$email_class = tribe( TEC\Tickets\Emails\Email\Ticket::class );
+
+			if ( ! $email_class->is_enabled() ) {
+				return $attachments;
+			}
+		}
+
+		if ( ! tribe_is_truthy( tribe_get_option( self::$option_add_event_ics, true ) ) ) {
+			return $attachments;
+		}
+
+		$post_id = $email_class->__get( 'post_id' );
+
+		if ( ! tribe_is_event( $post_id ) ) {
+			return $attachments;
+		}
+
+		$event = tribe_get_event( $post_id );
+
+		if ( empty( $event ) ) {
+			return $attachments;
+		}
+
+		$attachments = tribe( TEC_Email_Handler::class )->add_event_ics_to_attachments( $attachments, $post_id );
+
+		return $attachments;
+	}
+
+	/**
+	 * Maybe include event links.
+	 *
+	 * @since TBD
+	 *
+	 * @param \Tribe__Template $parent_template Event Tickets template object.
+	 *
+	 * @return void
+	 */
+	public function include_event_links( $parent_template ) {
+		$email_class = tribe( RSVP_Email::class );
+
+		// Bail early if the email class is not enabled.
+		if ( ! $email_class->is_enabled() ) {
+			return;
+		}
+
+		$use_ticket_email = tribe_get_option( $email_class->get_option_key( 'use-ticket-email' ), false );
+
+		if ( ! empty( $use_ticket_email ) ) {
+			$email_class = tribe( Ticket::class );
+
+			if ( ! $email_class->is_enabled() ) {
+				return;
+			}
+		}
+
+		if ( ! tribe_is_truthy( tribe_get_option( static::$option_add_event_links, true ) ) ) {
+			return;
+		}
+
+		$args = $parent_template->get_local_values();
+
+		if (
+			! empty( $args['email'] )
+			&& $args['email']->get_id() !== $email_class->get_id()
+		) {
+			return;
+		}
```

Comment:
I fixed this logic in https://github.com/the-events-calendar/the-events-calendar/pull/4226

In the case where RSVP is using Ticket settings, we're getting the wrong ID here. Just an FYI. It's fixed in the above PR.

Reply:
Lets fix that after either one is merged.

Reply:
It is fixed in #4226 . I was just mentioning it in case someone else caught it.